### ISSUE TITLE
Cloaking fails to cloak partial host names when host name has "." within it

### ISSUE THREAD

**Issue by bubthegreat** [inspircd/inspircd#1336]
Straightforward issue - cloaking breaks for any hostname that has a "." in it - so if my hostname is username@host.name.something.com, it will only cloak the username@host. and still exposes the .name.something.com.

from m_cloaking:

```
	std::string LastTwoDomainParts(const std::string &host)
	{
		int dots = 0;
		std::string::size_type splitdot = host.length();

		for (std::string::size_type x = host.length() - 1; x; --x)
		{
			if (host[x] == '.')
			{
				splitdot = x;
				dots++;
			}
			if (dots >= 3)
				break;
		}

		if (splitdot == host.length())
			return "";
		else
			return host.substr(splitdot);
	}
```
Seems like modifying this to if (dots >= 2) would be enough to still cloak appropriately, and would just "over cloak" the network, which doesn't seem to be an issue unless there are other modules that depend on the fully cloaked name.

The logic seems straightforward in that it looks for the last "." from the right, and then cloaks everything to the left of that location, when it should only do so for the first two so you cloak anything before domain.extension, but I may be missing something that this would break dependencies for, so bugging it for wiser minds to review.

**Comment by SISheogorath** [comment]
Just as a hint: The word `hostname` describes the host part of an FQDN(Full Qualified Domain Name) or to be more correct: FQHN (Full Qualified Host Name). The delimiter for a Domain name is always a `.`. in other words a Hostname can't contain a `.` by definition.

Anyways are right that the current way does show the 3rd domain part. So the function or the description of it is wrong. I'm pretty sure we can't fix this for `insp20` but for `master` aka InspIRCd 3.0 it should be possible :)

If you want to fix it yourself feel free to file a PR. Otherwise we'll take care of it :+1:

**Comment by SaberUK** [comment]
>I'm pretty sure we can't fix this for insp20 but for master aka InspIRCd 3.0 it should be possible :)

We need to retain link compat with 2.0 so in order to fix in master #1227 or something similar would be required.

**Comment by SISheogorath** [comment]
You're right. We can add a new config parameter to configure how many parts of the domain left as cleartext in mode "half". In full mode, it's ignored.

**Comment by SaberUK** [comment]
@bubthegreat This has been implemented in #1353.
